packages/components/src/components/domainproperties/dataset/models.ts

```diff
@@ -117,21 +133,47 @@ export class DatasetModel extends Record({
     }
 
     validManagedKeyField(): boolean {
-        if (this.keyPropertyId) {
+        if (this.keyPropertyName) {
             const domainFields = this.domain.fields;
 
             const allowedFieldTypes = domainFields
                 .filter(field => {
                     return field.dataType.isString() || field.dataType.isInteger();
                 })
                 .map(field => {
-                    return field.propertyId;
+                    return field.name;
                 })
                 .toList();
 
-            return allowedFieldTypes.contains(this.keyPropertyId);
+            return allowedFieldTypes.contains(this.keyPropertyName);
         } else {
             return false;
         }
     }
+
+    getDomainKind(): string {
+        if (getServerContext().moduleContext.study.timepointType === 'DATE') {
+            return 'StudyDatasetDate';
+        } else if (getServerContext().moduleContext.study.timepointType === 'VISIT') {
+            return 'StudyDatasetVisit';
+        }
+        return undefined;
+    }
+
+    getOptions(): Record<string, any> {
+        return produce(this, (draft: Draft<IDatasetModel>) => {
+            const model = draft;
+
+            delete model.exception;
+            delete model.domain;
```

Comment:
I assume that doing `delete draft.exception` and `delete draft.domain` might work the same here.